### PR TITLE
[MM-54599] Fix Servers bottom sheet

### PR DESCRIPTION
**Pull request by Paul-vrn**
#### Summary

Modified `BottomSheetFlatList` style in `app/screens/home/channel_list/servers/server_list/index.tsx` to add a bottom margin and prevent the "+Add server" bottom sheet from covering a server item.

Modified `onPress` function from `app/screens/home/channel_list/servers/index.tsx` to pre-calculate the "Your servers" Bottom sheet's height depending on the number of server items (and pushAlertText if present) up to a height of 60% of the screen. Above 60%, the sheet can be extended up to 80% of the screen.

#### Ticket Link

Fixes [https://github.com/mattermost/mattermost/issues/24668](https://github.com/mattermost/mattermost/issues/24668)
Jira [MM-54599](https://mattermost.atlassian.net/browse/MM-54599)

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [x] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on: Iphone 15, iOS 17.0 (using Simulator 15.0 on Mac)

#### Screenshots

one_server
two_servers
three_servers
four_servers

https://github.com/mattermost/mattermost-mobile/assets/78180340/79c6c44f-a9a6-4a3e-8de6-000686e35e64

#### Release Note

```release-note
NONE
```

[MM-54599]: https://mattermost.atlassian.net/browse/MM-54599?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/screens/home/channel_list/servers/servers_list/index.tsx [hunk 1]
```diff
@@ -4,8 +4,9 @@
 import {BottomSheetFlatList} from '@gorhom/bottom-sheet';
 import React, {useCallback, useMemo} from 'react';
 import {useIntl} from 'react-intl';
-import {FlatList, type ListRenderItemInfo, StyleSheet, View} from 'react-native';
+import {FlatList, StyleSheet, View, type ListRenderItemInfo} from 'react-native';
 
+import {BUTTON_HEIGHT} from '@app/screens/bottom_sheet';
```
Comment (enahum) [inline app/screens/home/channel_list/servers/servers_list/index.tsx:9]:
```suggestion
import {BUTTON_HEIGHT} from '@screens/bottom_sheet';
```

And sort the import

File: app/screens/home/channel_list/servers/index.tsx [hunk 2]
```diff
@@ -3,16 +3,16 @@
 
 import React, {useCallback, useEffect, useImperativeHandle, useRef, useState} from 'react';
 import {useIntl} from 'react-intl';
-import {StyleSheet} from 'react-native';
+import {Dimensions, StyleSheet} from 'react-native';
 import {useSafeAreaInsets} from 'react-native-safe-area-context';
 
+import {BUTTON_HEIGHT, TITLE_HEIGHT} from '@app/screens/bottom_sheet';
```
Comment (enahum) [inline app/screens/home/channel_list/servers/index.tsx:9]: Remove
```suggestion
```

File: app/screens/home/channel_list/servers/index.tsx [hunk 3]
```diff
@@ -3,16 +3,16 @@
...
 import ServerIcon from '@components/server_icon';
 import {useServerUrl} from '@context/server';
 import {useTheme} from '@context/theme';
 import {subscribeAllServers} from '@database/subscription/servers';
 import {subscribeUnreadAndMentionsByServer, type UnreadObserverArgs} from '@database/subscription/unreads';
 import {useIsTablet} from '@hooks/device';
-import {BUTTON_HEIGHT, TITLE_HEIGHT} from '@screens/bottom_sheet';
```
Comment (enahum) [inline app/screens/home/channel_list/servers/index.tsx:15]: Keep

File: app/screens/home/channel_list/servers/index.tsx [hunk 4]
```diff
@@ -116,12 +116,18 @@ const Servers = React.forwardRef<ServersRef>((_, ref) => {
                     <ServerList servers={registeredServers.current!}/>
                 );
             };
+            const maxScreenHeight = Math.ceil(0.6 * Dimensions.get('window').height);
+            const maxSnapPoint = Math.min(
+                maxScreenHeight,
+                bottomSheetSnapPoint(registeredServers.current.length, 75, bottom) + TITLE_HEIGHT + BUTTON_HEIGHT +
+                    (registeredServers.current.filter((s: ServersModel) => s.lastActiveAt).length * 42),
```
Comment (enahum) [inline app/screens/home/channel_list/servers/index.tsx:123]: Add the 42 to a constant name, same as the 75 to clearly identify what it means

**Review by enahum — CHANGES_REQUESTED** [review]

**Review by enahum — APPROVED** [review]

**Review by lindalumitchell — APPROVED** [review]
Verified on the PR builds on Android and iOS that after adding enough servers to cause the list to scroll, I can open the bottom sheet to view the servers, pull it up to expand the list taller, and scroll through the server list without issue. Also double-checked that functions from that list such as editing server name, removing server, and logging out of a server work as expected with no issues. 👍

**Pull request merged**